tests/Go/Instrument/Transformer/_files/php82-file-transformed.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Parser Reflection API
+ *
+ * @copyright Copyright 2016, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+declare(strict_types=1);
+
+namespace Go\ParserReflection\Stub;
+
+/**
+ * @see https://php.watch/versions/8.2/readonly-classes
+ */
+readonly class ClassWithPhp82ReadOnlyFlag
+{
+    public int $publicInt;
+}
+
+/**
+ * @see https://php.watch/versions/8.2/dnf-types
+ */
+class ClassWithPhp82DNFType
+{
+    private (JSONResponse&SuccessResponse)|HTMLResponse|string $respond;
+
+    public function __construct((JSONResponse&SuccessResponse)|HTMLResponse|string $respond)
+    {
+        $this->respond = $respond;
+    }
+
+    public function respond(): (JSONResponse&SuccessResponse)|HTMLResponse|string
+    {
+        return $this->respond;
+    }
+}
+
+/**
+ * @see https://php.watch/versions/8.2/null-false-types
+ * @see https://php.watch/versions/8.2/true-type
+ */
+class ClassWithPhp82NullFalseTypes
+{
+    private true $isTrue = true;
+    private false $isFalse = false;
+    private null $isNull = null;
+
+    public function returnsFalse(): false
+    {
+        return false;
+    }
+
+    public function returnsTrue(): true
+    {
+        return true;
+    }
+
+    public function returnsNullExplicitly(): null
+    {
+        return null;
+    }
+
+    public function acceptsTrue(true $acceptsTrue): void {}
+    public function acceptsFalse(false $acceptsFalse): void {}
+    public function acceptsNull(null $acceptsNull): void {}
+}
+
+/**
+ * @see https://php.watch/versions/8.2/constants-in-traits
+ */
+trait TraitWithPhp82Constant
+{
+    protected const CURRENT_VERSION = '2.6';
+    final protected const MIN_VERSION = '2.5';
+
+    protected function ensureVersion(): void
+    {
+        if (\Go\ParserReflection\Stub\ClassWithPhp82NullFalseTypes::CURRENT_VERSION < \Go\ParserReflection\Stub\ClassWithPhp82NullFalseTypes::MIN_VERSION) {
```

Comment:
on trait, the transformation seems incorrect, even on public constant, see 

- https://3v4l.org/uVC61#v8.2.18
- https://3v4l.org/YWIMP#v8.2.18

keep self seems the way to go: https://3v4l.org/aS44R#v8.2.18

Reply:
I tried to use `ParentConnectingVisitor` on  `SelfValueTransformer` to allow traverse up to verify it inside Trait, but it somehow can't get parent on target `self`, do you have idea to verify it ?

Reply:
I would better go with extra variable/check ` elseif ($node instanceof Trait_) {` and store this information as boolean flag `isInsideTrait` to stop transformation inside traits at all.

Reply:
sure, I updated to verify inside trait or enum, as the bug is happen on enum and trait https://github.com/goaop/framework/pull/504/commits/e13648e1bb9522c9911b56ec26a38f26259153df

Reply:
Ok, I found that the bug is due to `className` property re-use when single file consist of multiple `ClassLike`, I create a real patch for it https://github.com/goaop/framework/pull/504/commits/19c8ce81f13084f7be101d4a62daccaccf5574aa to make it null on `Trait_`